test/test_datapipe.py

```diff
@@ -1501,6 +1500,14 @@ def _create_dp(buffer_size):
                     multiprocessing_context=mp_ctx,
                     worker_init_fn=_worker_init_fn
                 )
+
+                # No seed
+                dl_res_ns = list(dl)
+                self.assertEqual(len(dl_res_ns), len(exp))
+                self.assertEqual(sorted(dl_res_ns), sorted(exp))
```

Comment:
Even without seed, all shuffer should have the same seed/state per worker process. Then, there won't be any data loss during iteration.